src/lava/magma/core/learning/learning_rule.py

```diff
@@ -144,13 +126,45 @@ def __init__(
             else np.random.randint(1, np.iinfo(np.int32).max)
         )
 
+        self._create_product_series()
+
+    def _validate_learning_rule_strings(self):
+        dw = None if self._dw_str is None else self._dw_str[5:]
+        dd = None if self._dd_str is None else self._dd_str[5:]
```

Comment:
See above, for `self.dd_str` and for `self.dt_str`.

Reply:
See answer above.